components/bounty/bounty-hero/controller.tsx

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from "next/router";
+
+import { useAppState } from "contexts/app-state";
+
+import { getIssueState } from "helpers/handleTypeIssue";
+
+import BountyHeroView from "./view";
+
+export default function BountyHeroController({
+  handleEditIssue,
+  isEditIssue,
+}: {
+  handleEditIssue?: () => void;
+  isEditIssue?: boolean;
+}) {
+  const router = useRouter();
+
+  const { state } = useAppState();
```

Comment:
necessary bounty state should be handed by page-controller or parent-controller, as we should avoid using app-state